clang/lib/Frontend/CompileJobCacheKey.cpp

```diff
@@ -56,18 +56,26 @@ clang::createCompileJobCacheKey(CASDB &CAS, ArrayRef<const char *> CC1Args,
 
 Optional<llvm::cas::CASID>
 clang::createCompileJobCacheKey(CASDB &CAS, DiagnosticsEngine &Diags,
-                                const CompilerInvocation &Invocation) {
+                                const CompilerInvocation &OriginalInvocation) {
+  CompilerInvocation InvocationForCacheKey(OriginalInvocation);
+  FrontendOptions &FrontendOpts = InvocationForCacheKey.getFrontendOpts();
+  DiagnosticOptions &DiagOpts = InvocationForCacheKey.getDiagnosticOpts();
```

Comment:
The offline feedback I provide is that it would be good just to have CompilerInvocation to create compilation cache key, which just hashing over all the options and each options knows what are the outputs and exclude those from hashing. Not sure how complicated that is.

Reply:
I removed the plain arguments variant with a [separate commit](https://github.com/apple/llvm-project/pull/5036/commits/3038da5863ffa80fdc1661929922c21c8c1f1b0e).

The only caller was the dependency scanning daemon, using it to do a check of whether the key is cached or not and track the cache hits number in a global variable. This particular use is debatable whether it provides enough value IMO.